spec/helpers/cd4pe_job_helper_spec.rb

```diff
@@ -0,0 +1,196 @@
+      expect(File.exists?(test_dir)).to be(false)
+  
+      # create dir and validate it exists
+      job_helper.make_working_dir(test_dir)
+      expect(File.exists?(test_dir)).to be(true)
+  
+      # attempt to create again to validate it does not throw
+      job_helper.make_working_dir(test_dir)
+    end
+  end
+
+  describe 'cd4pe_job_helper::get_docker_run_cmd' do
+    it 'Generates the correct docker run command.' do
+      test_manifest_type = "AFTER_JOB_SUCCESS"
+      test_docker_image = 'puppetlabs/yoloswag:10.0.1'
+      arg1 = '--testarg=woot'
+      arg2 = '--otherarg=hello'
+      arg3 = '--whatever=yolo'
+      user_specified_docker_run_args = [arg1, arg2, arg3]
+  
+      job_helper = PuppetX::Puppetlabs::CD4PEJobHelper.new(working_dir: @working_dir, docker_image: test_docker_image, docker_run_args: user_specified_docker_run_args)
+  
+      docker_run_command = job_helper.get_docker_run_cmd(test_manifest_type)
+      cmd_parts = docker_run_command.split(' ')
+  
+      expect(cmd_parts[0]).to eq('docker')
+      expect(cmd_parts[1]).to eq('run')
+      expect(cmd_parts[2]).to eq('--testarg=woot')
+      expect(cmd_parts[3]).to eq('--otherarg=hello')
+      expect(cmd_parts[4]).to eq('--whatever=yolo')
+      expect(cmd_parts[5]).to eq('-v')
+      expect(cmd_parts[6].end_with?("/#{File.basename(@working_dir)}/cd4pe_job/repo:/repo")).to be(true)
+      expect(cmd_parts[7]).to eq('-v')
+      expect(cmd_parts[8].end_with?("/#{File.basename(@working_dir)}/cd4pe_job/jobs/unix:/cd4pe_job")).to be(true)
+      expect(cmd_parts[9]).to eq('puppetlabs/yoloswag:10.0.1')
+      expect(cmd_parts[10]).to eq('/cd4pe_job/AFTER_JOB_SUCCESS')
+    end
+  end
+end
+
+describe 'cd4pe_job_helper::run_job' do
+
+  before(:all) do
+    @@logs = []
+    @working_dir = File.join(Dir.getwd, "test_working_dir")
+    cd4pe_job_dir = File.join(@working_dir, 'cd4pe_job')
+    jobs_dir = File.join(cd4pe_job_dir, 'jobs')
+    unix_dir = File.join(jobs_dir, 'unix')
+    @job_script = File.join(unix_dir, 'JOB')
+    @after_job_success_script = File.join(unix_dir, 'AFTER_JOB_SUCCESS')
+    @after_job_failure_script = File.join(unix_dir, 'AFTER_JOB_FAILURE')
+
+    Dir.mkdir(@working_dir)
+    Dir.mkdir(cd4pe_job_dir)
+    Dir.mkdir(jobs_dir)
+    Dir.mkdir(unix_dir)
+
+    File.write(@job_script, '')
+    File.chmod(0775, @job_script)
+    File.write(@after_job_success_script, '')
+    File.chmod(0775, @after_job_success_script)
+    File.write(@after_job_failure_script, '')
+    File.chmod(0775, @after_job_failure_script)
+  end
+
+  after(:all) do
+    FileUtils.remove_dir(@working_dir)
+  end
+
+  it 'Attempts to use docker when docker_image is specified during initialization' do
+    $stdout = StringIO.new
+    fake_docker_image = 'yolo/swag:for-the-lulz'
+    job_helper = PuppetX::Puppetlabs::CD4PEJobHelper.new(working_dir: @working_dir, docker_image: fake_docker_image)
+    output = job_helper.run_job
```

Comment:
Does Docker need to be installed to run these tests? If so, there are a couple of things that come to mind. What version of Docker do we support and should we write down the requirement somewhere?

Reply:
I removed the one test that required docker. Good call.